File: Backend/server/schemas/food.py
from typing import Optional, Any

from pydantic import BaseModel, Field
from .edamam import EdamamNutritionInfoSchema

class FoodSchema(BaseModel):
    food_id: Optional[int] = Field(None)
    name: str
    serving_size: float
    unit: str

    calories: float
    protein_g: float
    carbs_g: float
    fat_g: float

File: Frontend/lib/models/food.dart
class Food {
  final int? foodId;
  final String name;
  final double calories;
  final double proteinG;
  final double carbsG;
  final double fatG;
  final double servingSize;

  Food({
    this.foodId,
    required this.name,
    required this.calories,
    required this.proteinG,
    required this.carbsG,
    required this.fatG,
    required this.servingSize,
  });
}
